tests/ztest/base/src/main.c

```diff
@@ -82,7 +82,9 @@ static void rule_before_each(const struct ztest_unit_test *test, void *data)
 
 		zassert_equal_ptr(&rule_tests_fixture, data,
 				  "Data expected to point to rule_state");
-		zassert_equal(fixture->state, RULE_STATE_SETUP, "Unexpected state");
+		zassert_true((fixture->state == RULE_STATE_SETUP ||
+			    fixture->state == RULE_STATE_AFTER_EACH),
+			   "Unexpected state %u", fixture->state);
```

Comment:
I think we should add state here, what we're expecting is:
```
3x {
  setup
  before_each
  after_each
  before_each
  after_each
  before_each
  after_each
}
```

So this fails because we go back to `before_each`. I think we should add a count to the `rules_tests_fixture` and then assert based on both the state and the repeat count Kconfig